dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/PrivateDataManager.kt

```diff
@@ -301,4 +268,64 @@ class PrivateDataManager(
     fun getDocumentFromInMemoryStore(hash: String): ByteArray? {
         return documentInMemoryStorage[hash]
     }
+
+    /**
+     * Retrieves a private sme data object from the private storage
+     * @param dataId the dataId of the dataset to be retrieved
+     * @param correlationId the correlationId of the request
+     */
+    fun getPrivateDataSet(dataId: String, correlationId: String): StorableDataSet {
+        return dataManagerUtils.getDataSet(
+            dataId, DataType.of(SmeData::class.java), correlationId,
+            ::getDataFromCacheOrStorageService,
+        )
+    }
+    private fun getDataFromCacheOrStorageService(dataId: String, correlationId: String): String {
+        return jsonDataInMemoryStorage[dataId] ?: dataManagerUtils
+            .getDataFromStorageService(
+                dataId, correlationId,
+                ::getPrivateData,
+            )
+    }
+    private fun getPrivateData(dataId: String, correlationId: String): String {
+        return storageClientUtils.getDataByIdFromStorageClient(dataId, correlationId)
+    }
```

Comment:
Analog zum DataManager: Erscheint mir zu kompliziert.
Habe das Gefühl, das kann man schlanker und klarer umsetzen.
Stell dir vor jmd Neues kommt auf das Projekt und liest "getPrivateDataSet()" und "getPrivateData()" und muss den Unterschied verstehen.

Reply:
Ok gerne.   Offenbar ist das in main auch schon seltsam.